assets/component-visual-display.css

```diff
@@ -0,0 +1,107 @@
+.visual-display {
+  --visual-display__size: min(2.4rem, 100%);
+  position: relative;
+  width: var(--visual-display__size);
+  border: 0.1rem solid rgba(var(--color-foreground), 0.2);
+  aspect-ratio: 1/1;
+}
+
+.visual-display.empty {
+  border-style: dashed;
+}
+
+.visual-display--style-swatch {
+  --visual-display__size: min(2.4rem, 100%);
+
+  border-radius: 100%;
+  overflow: hidden;
+}
+
+.visual-display--style-thumbnail {
+  --visual-display__size: min(9rem, 100%);
+
+  border-radius: var(--buttons-radius);
+  overflow: hidden;
+}
+
+.visual-display-parent {
+  outline-offset: 0.2rem;
+}
+
+.visual-display-parent--thumbnail {
+  outline: 0.05rem solid rgba(var(--color-foreground), 0.75);
+}
+
+/* Hover, active, and focus states */
+:where(
+    .visual-display-parent:hover .visual-display--style-swatch,
+    .visual-display-parent.active .visual-display--style-swatch,
+    .visual-display-parent:has(:focus-visible) .visual-display--style-swatch,
+    .visual-display-parent--thumbnail:hover,
+    .visual-display-parent--thumbnail.active,
+    .visual-display-parent--thumbnail:has(:focus-visible)
+  ) {
+  outline: 0 solid rgb(var(--color-foreground));
+}
+
+/* Active state */
+.visual-display-parent.active .visual-display--style-swatch,
+.visual-display-parent--thumbnail.active {
+  outline-width: 0.1rem;
+}
+
+/* Hover state */
+.visual-display-parent:hover .visual-display--style-swatch,
+.visual-display-parent--thumbnail:hover {
+  outline-width: 0.15rem;
+}
+
+/* Focus state */
+.visual-display-parent:has(:focus-visible) .visual-display--style-swatch,
+.visual-display-parent--thumbnail:has(:focus-visible) {
+  outline-width: 0.15rem;
+  box-shadow: 0 0 0 0.55rem rgb(var(--color-background)), 0 0 0 0.85rem rgba(var(--color-foreground), 0.5),
+    0 0 1.15rem 0.1rem rgba(var(--color-foreground));
+}
+
+/* Focus state for older browsers */
+@supports not selector(:has(a, b)) {
```

Comment:
What is `a` and `b` here? 👀

Reply:
Thanks for the clarification. Im not sure if we need this here 🤔 Is there a reason why we are using `focus-within`

In Dawn, we have used `focus` and `focus-visible` historically

https://github.com/search?q=repo%3AShopify%2Fdawn+%3Afocus&type=code

Reply:
So this is a pretty unique scenario, but I essentially want to know if _anything_ inside of `visual-display-parent` has the `:focus-visible` (thus only visible if you use your keyboard to focus, not clicking, and then style the `.visual-display` inside of it. There isn't a css pseudo class that allows this, and there's a pretty good [article](https://larsmagnus.co/blog/focus-visible-within-the-missing-pseudo-class) about it.

Essentially I have this class `.visual-display-parent:has(:focus-visible) .visual-display--style-swatch` giving us the preferred css selector, then this `@supports not` style will be the fallback for browsers that don't yet support `:has`, giving them the less ideal `:focus-within` experience.

Here's a [sandbox](https://codesandbox.io/s/holy-fast-mdnhsq?file=/src/styles.css) and a [video](https://screenshot.click/12-40-j67lj-2tk2h.mp4) showing the difference!